salt/addons/deploy_addons.sh

```diff
@@ -14,11 +14,17 @@ abort() { echo "FATAL: $@" ; exit 1 ; }
 NS_ARGS="--namespace=kube-system"
 SRV_ARGS="--server=http://127.0.0.1:8080"
 
+# Each kubectl command is potentially ran twice, as `apply -f` is not truly
+# idempotant. The "create or update" logic is client side, so there is a check
+# + set race condition. If it fails, we try it again, if it fails again, we will
+# fail the entire execution of this script. This works around the "create or update"
+# check+set race.
+
 echo "Creating kube-system namespace..."
 # use kubectl to create kube-system namespace
 NAMESPACE=`eval "$KUBECTL_PATH $SRV_ARGS get namespaces | grep kube-system | cat"`
 if [ ! "$NAMESPACE" ]; then
-	$KUBECTL_PATH $SRV_ARGS create -f "$DIR/namespace.yaml"
+	$KUBECTL_PATH $SRV_ARGS apply -f "$DIR/namespace.yaml" || $KUBECTL_PATH $SRV_ARGS apply -f "$DIR/namespace.yaml"
```

Comment:
Maybe you should refactor this to a common function:

```bash
kubectl_apply() {
    $KUBECTL_PATH $SRV_ARGS apply -f $1" || $KUBECTL_PATH $SRV_ARGS apply -f $1
}
```

and you could even loop until the `kubectl` succeeds.